src/sentry/issues/escalating.py

```diff
@@ -333,6 +333,14 @@ def manage_issue_states(
             )
             if data and activity_data:
                 data.update(activity_data)
+            group.save(update_fields=["status", "substatus"])
```

Comment:
I think we might be able to remove the `group.save()` calls. The conditional update should take care of updating the db and setting the `status` and `substatus` fields should update the model for the group if any code downstream is gonna use it.